drivers/clock_control/clock_control_silabs_siwx91x.c

```diff
@@ -177,6 +187,14 @@ static enum clock_control_status siwx91x_clock_get_status(const struct device *d
 
 static int siwx91x_clock_init(const struct device *dev)
 {
+#ifdef CONFIG_PM
+	sl_power_peripheral_t peripheral_config = {0};
+	sl_power_ram_retention_config_t ram_configuration = {
+		.configure_ram_banks = false,
+		.m4ss_ram_size_kb = M4SS_RAM_SIZE_KB,
+		.ulpss_ram_size_kb = 4,
+	};
+#endif
```

Comment:
Remove the `#ifdef` around these declaration (add attribute `__maybe_unused` if the CI complains). So, you will be able to replace `#ifdef CONFIG_PM` by `if (IS_ENABLED(CONFIG_PM))` below.